File: smartlist/src/main/java/com/smartlist/module/file/FileResponseCodes.java
package com.smartlist.module.file;

import com.smartlist.utils.AppResponseCodes;
import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

public class FileResponseCodes extends AppResponseCodes {

    private FileResponseCodes(){}

    public static final String FILE_UNSUPPORTED_DESCRIPTION = "unsupported file type";
    public static final String FILE_EMPTY_DESCRIPTION = "file missing";
    public static final String INVALID_FILE_ID_DESCRIPTION = "invalid file id";


    public static ResponseStatusException FILE_UNSUPPORTED = new ResponseStatusException(HttpStatus.BAD_REQUEST, FILE_UNSUPPORTED_DESCRIPTION);

    public static ResponseStatusException FILE_EMPTY = new ResponseStatusException(HttpStatus.NOT_FOUND, FILE_EMPTY_DESCRIPTION);

    public static ResponseStatusException INVALID_FILE_ID = new ResponseStatusException(HttpStatus.NOT_FOUND, INVALID_FILE_ID_DESCRIPTION);



}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/file/file.response.codes.ts
import { BadRequestException, NotFoundException } from '@nestjs/common';
import { AppResponseCodes } from 'src/app.response.codes';

export class FileResponseCodes extends AppResponseCodes {
  public static FILE_UNSUPPORTED: any = new BadRequestException(
    'Unsupported file type',
  );

  public static FILE_EMPTY: any = new NotFoundException('file missing');

  public static INVALID_FILE_ID: any = new NotFoundException('invalid file id');
}
